### DISCUSSION TITLE
Everything but chrome breaks in production

### DISCUSSION THREAD

**Discussion by jamesbalcombe83** [go-pkgz/auth#185] [Q&A]
I can't find the root cause of this myself so far, so reaching out for some guidance/help.

I'm only using google for the time being in my application, which in my dev environment, including final containerized application, works absolutely fine in every browser i've tried for login processes. However the actual deployed application only works in chrome.

When trying to login in firefox/edge/etc. the process gets to the point of this call `https://accounts.google.com/o/oauth2/auth` and rather than returning 200, gives back 302, which the causes the callback to return a 500 error 
```
{
    "error": "failed to get token"
}
```
If this were in dev I'd trace the code and find the issue that way, but I cannot replicate the problem there. 
The only thing I can think of is that the nginx config on the server which directs to the exposed docker port is doing something, but then I would expect that to affect all browsers, rather than everything but chrome.

This is the nginx conf for the domain
```
server {
        #root /var/www/your_domain/html;
        #index index.html index.htm index.nginx-debian.html;

        server_name my_domain.com www.my_domain.com;

        location / {
                proxy_set_header X-Forwarded-For $remote_addr;
                proxy_set_header   Host $http_host;
                proxy_pass         "http://127.0.0.1:8080";
        }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/my_domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/my_domain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}

server {
    if ($host = www.my_domain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    if ($host = my_domain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

        listen 80;
        listen [::]:80;

        server_name my_domain.com www.my_domain.com;

    return 404; # managed by Certbot
}
```

Any suggestions or ideas as to what might be causing this, or how to investigate further?

**Comment by paskal** [comment]
What was the solution in your case?

**Reply by jamesbalcombe83** [reply]
Rather embarrassingly it was to stop being an idiot and write the URL correctly in the search bar. I forgot I hadn't fully set up mydomain.com vs www.mydomain.com in google console
